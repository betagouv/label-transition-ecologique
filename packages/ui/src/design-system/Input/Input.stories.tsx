import {Meta, StoryObj} from '@storybook/react';
import {action} from '@storybook/addon-actions';

import {Input} from './Input';

const meta: Meta<typeof Input> = {
  title: 'Design System/Input',
  component: Input,
};

export default meta;

type Story = StoryObj<typeof Input>;

/** Sans aucune props renseignée. */
export const Default: Story = {
  args: {},
};

/** Avec une aide à la saisie quand le champ est vide */
export const AvecPlaceholder: Story = {
  args: {
    placeholder: 'Saisir une valeur',
    'data-test': 'ok',
  },
};

/** Avec une valeur */
export const AvecValeur: Story = {
  args: {
    value: 'plop',
    onChange: action('onChange'),
  },
};

/** Avec icône */
export const AvecIcone: Story = {
  args: {
    placeholder: 'placeholder',
    icon: {value: 'search-line'},
  },
};

/** Avec bouton icône */
export const AvecIconeBouton: Story = {
  args: {
    placeholder: 'placeholder',
    icon: {
      buttonProps: {
        icon: 'search-line',
        onClick: action('onClick'),
        title: 'Rechercher',
      },
    },
  },
};

/** Avec un texte à la place de l'icône */
export const AvecIconeTexte: Story = {
  args: {
    placeholder: 'placeholder',
    icon: {text: 'TTC'},
  },
};

/** Variante en fonction de l'état */
export const AvecBordsColorés: Story = {
  args: {
    placeholder: 'placeholder',
    state: 'error',
    icon: {
      buttonProps: {
        icon: 'search-line',
      },
    },
  },
};

/** Variante small */
export const VarianteSmall: Story = {
  args: {
    size: 'sm',
    placeholder: 'placeholder',
  },
};

/** Variante small et bouton icône */
export const VarianteSmallEtBouton: Story = {
  args: {
    size: 'sm',
    placeholder: 'placeholder',
    icon: {
      buttonProps: {
        icon: 'search-line',
        onClick: action('onClick'),
        title: 'Rechercher',
      },
    },
  },
};

/** Variante small et texte à la place de l'icône */
export const VarianteSmallIconeTexte: Story = {
  args: {
    size: 'sm',
    placeholder: 'placeholder',
    icon: {text: 'TTC'},
  },
};
