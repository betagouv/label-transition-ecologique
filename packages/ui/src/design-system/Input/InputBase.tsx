import {Ref, forwardRef} from 'react';
import classNames from 'classnames';
import {Icon, IconValue} from '@design-system/Icon';
import {DefaultButtonProps} from '@design-system/Button/types';
import {Button} from '@design-system/Button';
import {FieldState, stateToTextColor} from '@design-system/Field';
import {preset} from '@tailwind-preset';

// variantes de taille
export type InputSize = 'md' | 'sm';

// types de champ accepté
export type InputType = 'text' | 'password' | 'date' | 'search' | 'tel';

// couleur des bordures en fonction du `state`
const stateToBorderColor: Record<FieldState, string> = {
  default: 'border-grey-4',
  disabled: 'border-grey-4',
  info: 'border-info-1',
  error: 'border-error-1',
  success: 'border-success-1',
  warning: 'border-warning-1',
};

export type InputBaseProps = Omit<
  React.ComponentPropsWithoutRef<'input'>,
  'type'
> & {
  /** Type de saisie */
  type?: InputType;
  /** Valeur courante du champ */
  value?: string;
  /** Taille d'affichage */
  displaySize?: InputSize;
  /** Id pour les tests */
  'data-test'?: string;
  /** Contenu optionnel pour la zone d'icône à droite du champ */
  icon?: IconContent;
  /** Variante en fonction d'un état */
  state?: FieldState;
  /** Pour styler le container */
  containerClassname?: string;
};

type IconContent =
  // affiche un composant `Icon`
  | {value?: IconValue}
  // affiche un texte
  | {text?: string}
  // affiche un composant `Button`
  | {buttonProps?: Omit<DefaultButtonProps, 'variant' | 'size' | 'className'>};

/**
 * Composant de base pour les champs de saisie.
 */
export const InputBase = forwardRef(
  (props: InputBaseProps, ref?: Ref<HTMLInputElement>) => {
    const {
      className,
      containerClassname,
      type = 'text',
      displaySize = 'md',
      icon,
      state,
      ...remainingProps
    } = props;

    const borderColor = stateToBorderColor[state] || stateToBorderColor.default;

    return (
      <div
        className={classNames(
          'inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5',
          borderColor,
          containerClassname
        )}
      >
        <input
          type={type}
          ref={ref}
          className={classNames(
            'grow text-grey-8 px-4 outline-none',
            {
              'text-sm py-2': displaySize === 'sm',
              'text-md py-3': displaySize === 'md',
              'border-r border-solid': !!icon,
              [borderColor]: !!icon,
            },
            className
          )}
          {...remainingProps}
        />
        <InputIconContent {...props} />
      </div>
    );
  }
);

/**
 * Affiche le contenu de la zone icône/texte/bouton à droite du champ.
 */
const InputIconContent = ({
  disabled,
  icon,
  state,
  displaySize: size = 'md',
}: InputBaseProps) => {
  if (!icon) {
    return;
  }

  const iconColor = disabled
    ? 'text-grey-6'
    : stateToTextColor[state] || 'text-primary';

  /** icône */
  if ('value' in icon) {
    return (
      <Icon
        className={classNames('self-center min-w-[3rem]', iconColor)}
        icon={icon.value}
        size={size}
      />
    );
  }

  /** "icône" texte  */
  if ('text' in icon) {
    return (
      <span
        className={classNames(
          'self-center text-xs text-center font-bold min-w-[3rem]',
          iconColor
        )}
      >
        {icon.text}
      </span>
    );
  }

  /** icône bouton */
  if ('buttonProps' in icon) {
    return (
      <Button
        variant="white"
        size={size}
        disabled={disabled}
        className={classNames('rounded-none border-none')}
        // change la couleur de l'icône du bouton pour refléter le `state`
        // TODO: enlever les !important de `Button` pour pouvoir styler le bouton par le className
        style={{color: preset.theme.extend.colors[state]?.['1']}}
        {...icon.buttonProps}
      />
    );
  }
};
