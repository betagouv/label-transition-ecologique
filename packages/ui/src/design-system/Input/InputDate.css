/** cache l'icône du sélecteur de date (fonctionne uniquement pour chrome et probablement edge mais pas pour firefox ni safari) */
.input-date input[type='date']::-webkit-calendar-picker-indicator {
  display: none;
}
