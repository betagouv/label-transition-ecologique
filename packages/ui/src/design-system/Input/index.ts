export * from './Input';
export * from './InputDate';
export * from './InputPassword';
