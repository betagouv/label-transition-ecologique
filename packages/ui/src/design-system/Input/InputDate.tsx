import {useRef} from 'react';
import classNames from 'classnames';
import {InputBase, InputBaseProps} from './InputBase';

export type InputDateProps = Omit<InputBaseProps, 'icon' | 'type'>;

/**
 * Affiche un champ de saisie date
 */
export const InputDate = ({className, ...props}: InputDateProps) => {
  const inputRef = useRef<HTMLInputElement>(null);

  return (
    <InputBase
      type="date"
      ref={inputRef}
      className={classNames(
        /** cache l'icône du sélecteur de date (fonctionne uniquement pour chrome et probablement edge mais pas pour firefox ni safari) */
        '[&::-webkit-calendar-picker-indicator]:hidden',
        className
      )}
      icon={{
        buttonProps: {
          icon: 'calendar-line',
          onClick: () => {
            inputRef.current?.showPicker();
          },
        },
      }}
      {...props}
    />
  );
};
