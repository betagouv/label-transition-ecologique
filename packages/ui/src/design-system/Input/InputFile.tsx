import { Ref, forwardRef } from 'react';
import classNames from 'classnames';
import { InputBaseProps } from './InputBase';
import { FieldState } from '../Field';

const stateToBorderColor: Record<FieldState, string> = {
  default: 'border-grey-4',
  disabled: 'border-grey-4',
  info: 'border-info-1',
  error: 'border-error-1',
  success: 'border-success-1',
  warning: 'border-warning-1',
};

export type InputFileProps = Omit<InputBaseProps, 'icon' | 'type'>;

/**
 * Affiche un champ de s√©lection d'un fichier
 * TODO: drag and drop des fichiers
 */

export const InputFile = forwardRef(
  (
    {
      displaySize = 'sm',
      state,
      containerClassname,
      ...remainingProps
    }: InputFileProps,
    ref?: Ref<HTMLInputElement>
  ) => {
    const borderColor =
      state !== undefined ? stateToBorderColor[state] : stateToBorderColor.info;

    return (
      <div
        className={classNames(
          'bg-grey-1 border border-dashed rounded-lg flex flex-col justify-center items-center',
          {
            'p-4': displaySize === 'sm',
            'p-5': displaySize === 'md',
          },
          borderColor,
          containerClassname
        )}
      >
        <input
          ref={ref}
          id="input-file"
          type="file"
          className="invisible w-0 h-0"
          {...remainingProps}
        />
        <label
          htmlFor="input-file"
          className={classNames(
            'w-fit m-0 border-solid cursor-pointer rounded-lg border font-bold text-primary-7 hover:text-primary-8 bg-primary-0 hover:!bg-primary-1 border-primary-7 hover:!border-primary-8',
            {
              'text-xs py-2 px-5': displaySize === 'sm',
              'text-sm py-2.5 px-6': displaySize === 'md',
            }
          )}
        >
          Choisir un fichier
        </label>
      </div>
    );
  }
);
