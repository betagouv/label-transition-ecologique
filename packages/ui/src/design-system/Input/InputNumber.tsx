import {Ref, forwardRef, useEffect, useState} from 'react';
import {NumericFormat, NumericFormatProps} from 'react-number-format';
import {IconContent, InputBase, InputBaseProps} from './InputBase';

export type InputNumberProps = Omit<NumericFormatProps, 'type' | 'value'> & {
  value: InputBaseProps['value'];
  /** Type de saisie numérique attendue */
  numType?: 'float' | 'int';
  /** Contenu optionnel pour la zone d'icône à droite du champ */
  icon?: InputBaseProps['icon'];
  /** Pour styler le container */
  containerClassname?: InputBaseProps['containerClassname'];
};

const DECIMAL_SEP = ',';
const THOUSAND_SEP = ' ';

/**
 * Affiche un champ de saisie numérique.
 *
 * La saisie des caractères non valides est ignorée.
 * La valeur est formatée automatiquement avec les séparateurs de milliers et de décimales FR.
 */
export const InputNumber = forwardRef(
  (
    {numType = 'int', defaultValue, value, ...remainingProps}: InputNumberProps,
    ref?: Ref<HTMLInputElement>
  ) => {
    const [currentValue, setCurrentValue] = useState(defaultValue);

    useEffect(() => {
      setCurrentValue(value);
    }, [value]);

    return (
      <NumericFormat
        customInput={InputBase}
        getInputRef={ref}
        thousandSeparator={THOUSAND_SEP}
        decimalSeparator={DECIMAL_SEP}
        onPaste={e => {
          // rend la chaîne copiée compatible avec le `decimalSeparator` spécifié
          const data = e.clipboardData.getData('text/plain');
          if (typeof data === 'string' && data.includes('.')) {
            e.preventDefault();
            setCurrentValue(data.replace('.', DECIMAL_SEP));
          }
        }}
        type="text"
        value={currentValue}
        {...(numType === 'int' ? {decimalScale: 0} : {})}
        {...remainingProps}
      />
    );
  }
);
