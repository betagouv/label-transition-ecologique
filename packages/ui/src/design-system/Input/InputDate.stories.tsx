import {Meta, StoryObj} from '@storybook/react';
import {action} from '@storybook/addon-actions';
import {InputDate} from './InputDate';

const meta: Meta<typeof InputDate> = {
  title: 'Design System/InputDate',
  component: InputDate,
};

export default meta;

type Story = StoryObj<typeof InputDate>;

/** Sans aucune props renseignée. */
export const Default: Story = {
  args: {},
};

/** Avec une valeur renseignée. */
export const AvecValeur: Story = {
  args: {
    value: '2024-01-17',
    onChange: action('onChange'),
  },
};
