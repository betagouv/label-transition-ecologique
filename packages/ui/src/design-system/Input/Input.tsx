import {Ref, forwardRef} from 'react';
import {InputBase, InputBaseProps} from './InputBase';
import {InputDate, InputDateProps} from './InputDate';
import {InputSearch, InputSearchProps} from './InputSearch';
import {InputPassword, InputPasswordProps} from './InputPassword';
import {InputNumber, InputNumberProps} from './InputNumber';
import {InputPattern, InputPatternProps} from './InputPattern';
import {InputOTP, InputOTPProps} from './InputOTP';
import {InputTel, InputTelProps} from './InputTel';

type InputProps =
  | (Omit<InputBaseProps, 'type'> & {type: 'text'})
  | (InputNumberProps & {type: 'number'})
  | (InputDateProps & {type: 'date'})
  | (InputSearchProps & {type: 'search'})
  | (InputPasswordProps & {type: 'password'})
  | (InputPatternProps & {type: 'pattern'})
  | (InputTelProps & {type: 'tel'})
  | (InputOTPProps & {type: 'otp'});

/**
 * Affiche un champ de saisie, éventuellement combiné à une zone d'icône (ou de
 * texte) à droite du champ.
 * Toutes les props de l'élément HTML `input` sont acceptées à l'exception des changements suivants :
 * - type : restreint à un sous-ensemble des types standards
 */
export const Input = forwardRef(
  ({type = 'text', ...props}: InputProps, ref?: Ref<HTMLInputElement>) => {
    if (type === 'date') {
      return <InputDate {...(props as InputDateProps)} ref={ref} />;
    }

    if (type === 'search') {
      return <InputSearch {...(props as InputSearchProps)} ref={ref} />;
    }

    if (type === 'password') {
      return <InputPassword {...(props as InputPasswordProps)} ref={ref} />;
    }

    if (type === 'number') {
      return <InputNumber {...(props as InputNumberProps)} ref={ref} />;
    }

    if (type === 'pattern') {
      return <InputPattern {...(props as InputPatternProps)} ref={ref} />;
    }

    if (type === 'otp') {
      return <InputOTP {...(props as InputOTPProps)} ref={ref} />;
    }

    if (type === 'tel') {
      return <InputTel {...(props as InputTelProps)} ref={ref} />;
    }

    return <InputBase {...(props as InputBaseProps)} ref={ref} />;
  }
);
