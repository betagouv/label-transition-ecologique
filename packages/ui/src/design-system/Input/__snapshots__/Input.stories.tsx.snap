// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design System/Input AvecBordsColor√©s smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-error-1">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-error-1"
         placeholder="placeholder"
         value
  >
  <button class="w-fit border-solid font-bold group cursor-pointer rounded-lg border text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none"
          style="color: rgb(245, 91, 91);"
  >
    <div class="flex items-center gap-2">
      <span class="ri-search-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
      </span>
    </div>
  </button>
</div>
`;

exports[`Design System/Input AvecIcone smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4"
         placeholder="placeholder"
         value
  >
  <span class="ri-search-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] self-center min-w-[3rem] text-primary">
  </span>
</div>
`;

exports[`Design System/Input AvecIconeBouton smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4"
         placeholder="placeholder"
         value
  >
  <button title="Rechercher"
          class="w-fit border-solid font-bold group cursor-pointer rounded-lg border text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none"
  >
    <div class="flex items-center gap-2">
      <span class="ri-search-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
      </span>
    </div>
  </button>
</div>
`;

exports[`Design System/Input AvecIconeBoutonEtDesactive smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4"
         disabled
         placeholder="placeholder"
         value
  >
  <button title="Rechercher"
          disabled
          class="w-fit border-solid font-bold group cursor-not-allowed rounded-lg border !text-primary-4 !bg-white !border-white p-3.5 rounded-none border-none"
  >
    <div class="flex items-center gap-2">
      <span class="ri-search-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-4">
      </span>
    </div>
  </button>
</div>
`;

exports[`Design System/Input AvecIconeEtDesactive smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4"
         disabled
         placeholder="placeholder"
         value
  >
  <span class="ri-search-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] self-center min-w-[3rem] text-grey-6">
  </span>
</div>
`;

exports[`Design System/Input AvecIconeTexte smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4"
         placeholder="placeholder"
         value
  >
  <span class="self-center text-xs text-center font-bold min-w-[3rem] text-primary">
    TTC
  </span>
</div>
`;

exports[`Design System/Input AvecIconeTexteEtDesactive smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4"
         disabled
         placeholder="placeholder"
         value
  >
  <span class="self-center text-xs text-center font-bold min-w-[3rem] text-grey-6">
    TTC
  </span>
</div>
`;

exports[`Design System/Input AvecPlaceholder smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-md py-3"
         placeholder="Saisir une valeur"
         data-test="ok"
         value
  >
</div>
`;

exports[`Design System/Input AvecValeur smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-md py-3"
         value="plop"
  >
</div>
`;

exports[`Design System/Input Default smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-md py-3"
         value
  >
</div>
`;

exports[`Design System/Input TypeDate smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="date"
         class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4 [&amp;::-webkit-calendar-picker-indicator]:hidden"
         value
  >
  <button class="w-fit border-solid font-bold group cursor-pointer rounded-lg border text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none">
    <div class="flex items-center gap-2">
      <span class="ri-calendar-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
      </span>
    </div>
  </button>
</div>
`;

exports[`Design System/Input TypeDateAvecValeur smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="date"
         class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4 [&amp;::-webkit-calendar-picker-indicator]:hidden"
         value="2024-01-17"
  >
  <button class="w-fit border-solid font-bold group cursor-pointer rounded-lg border text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none">
    <div class="flex items-center gap-2">
      <span class="ri-calendar-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
      </span>
    </div>
  </button>
</div>
`;

exports[`Design System/Input TypeNumberFloat smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-md py-3"
         value
         inputmode="numeric"
  >
</div>
`;

exports[`Design System/Input TypeNumberFloatAvecValeur smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-md py-3"
         value="12 345,67"
         inputmode="numeric"
  >
</div>
`;

exports[`Design System/Input TypeNumberInt smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-md py-3"
         value
         inputmode="numeric"
  >
</div>
`;

exports[`Design System/Input TypeNumberIntAvecValeur smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-md py-3"
         value="12 345"
         inputmode="numeric"
  >
</div>
`;

exports[`Design System/Input TypePassword smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="password"
         class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4"
         value
  >
  <button title="Voir le mot de passe"
          class="w-fit border-solid font-bold group cursor-pointer rounded-lg border text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none"
  >
    <div class="flex items-center gap-2">
      <span class="ri-eye-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
      </span>
    </div>
  </button>
</div>
`;

exports[`Design System/Input TypePasswordAvecValeur smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="password"
         class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4"
         value="hyper secret"
  >
  <button title="Voir le mot de passe"
          class="w-fit border-solid font-bold group cursor-pointer rounded-lg border text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none"
  >
    <div class="flex items-center gap-2">
      <span class="ri-eye-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
      </span>
    </div>
  </button>
</div>
`;

exports[`Design System/Input TypeSearch smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="search"
         class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4 search-reset:appearance-none search-reset:w-5 search-reset:h-5 search-reset:bg-close-circle-fill"
         value
  >
  <button title="Rechercher"
          disabled
          class="w-fit border-solid font-bold group cursor-not-allowed rounded-lg border !text-primary-4 !bg-white !border-white p-3.5 rounded-none border-none"
  >
    <div class="flex items-center gap-2">
      <span class="ri-search-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-4">
      </span>
    </div>
  </button>
</div>
`;

exports[`Design System/Input TypeSearchAvecValeur smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="search"
         class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4 search-reset:appearance-none search-reset:w-5 search-reset:h-5 search-reset:bg-close-circle-fill"
         value="valeur recherch√©e"
  >
  <button title="Rechercher"
          class="w-fit border-solid font-bold group cursor-pointer rounded-lg border text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none"
  >
    <div class="flex items-center gap-2">
      <span class="ri-search-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
      </span>
    </div>
  </button>
</div>
`;

exports[`Design System/Input TypeSearchLoading smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="search"
         class="grow text-grey-8 px-4 outline-none text-md py-3 border-r border-solid border-grey-4 search-reset:appearance-none search-reset:w-5 search-reset:h-5 search-reset:bg-close-circle-fill"
         value="valeur recherch√©e"
  >
  <button title="Rechercher"
          class="w-fit border-solid font-bold group cursor-pointer rounded-lg border text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-3.5 rounded-none border-none"
  >
    <div class="flex items-center gap-2">
      <span class="ri-loader-4-line animate-spin font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
      </span>
    </div>
  </button>
</div>
`;

exports[`Design System/Input VarianteSmall smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-sm py-2"
         placeholder="placeholder"
         value
  >
</div>
`;

exports[`Design System/Input VarianteSmallEtBouton smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-sm py-2 border-r border-solid border-grey-4"
         placeholder="placeholder"
         value
  >
  <button title="Rechercher"
          class="w-fit border-solid font-bold group cursor-pointer rounded-lg border text-primary-7 hover:!text-primary-8 bg-white hover:!bg-primary-1 border-white hover:!border-primary-1 p-2.5 rounded-none border-none"
  >
    <div class="flex items-center gap-2">
      <span class="ri-search-line font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem] fill-primary-7 group-hover:fill-primary-8">
      </span>
    </div>
  </button>
</div>
`;

exports[`Design System/Input VarianteSmallIconeTexte smoke-test 1`] = `
<div class="inline-flex items-stretch border border-solid rounded-lg bg-grey-1 overflow-hidden focus-within:border-primary-5 border-grey-4">
  <input type="text"
         class="grow text-grey-8 px-4 outline-none text-sm py-2 border-r border-solid border-grey-4"
         placeholder="placeholder"
         value
  >
  <span class="self-center text-xs text-center font-bold min-w-[3rem] text-primary">
    TTC
  </span>
</div>
`;
